File: javascript/problem3.js
var primeFactors = [];
var findPrime = 600851475143;

var isPrime = function(num) {
	var prime;

	if(num==2) {
		return true;
	} else {
		for(var i=2; i<num; i++) {
			if (num%i != 0 && prime != false) {
				prime = true;
			} else {
				return false;
			}
		}
	}

	return prime;
}

var i = 0;

while(findPrime>0 && i<=findPrime) {
	if(isPrime(i) && findPrime%i == 0) {
		primeFactors.push(i);
		findPrime = findPrime/i;
	}
	i++;
}

console.log(primeFactors[primeFactors.length -1]);

File: ruby/problem3.rb
def prime(num)
	i = num
	prime = nil

	while i >= 1
		if num%i === 0 && num != i && i != 1
			return false
		elsif num === 2 || num === 1
			return false
		end

		i -= 1
	end

	return prime = true
end

input = 600851475143
i = 2
largestPrime = 0;

while i <= input
	if input%i === 0 && prime(i)
		largestPrime = i
		input /= i
	end

	i += 1
end

puts largestPrime